analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/TupleBinary.cs

```diff
@@ -0,0 +1,88 @@
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using SonarAnalyzer.SymbolicExecution.Constraints;
+
+namespace SonarAnalyzer.SymbolicExecution.Roslyn.OperationProcessors;
+
+internal sealed class TupleBinary : BinaryBase<ITupleBinaryOperationWrapper>
+{
+    protected override SymbolicConstraint BoolConstraintFromOperation(ProgramState state, ITupleBinaryOperationWrapper operation, bool isLoopCondition, int visitCount)
+    {
+        var tupleElements = TupleElements(operation.LeftOperand, operation.RightOperand);
+        var allEqual = true;
+        foreach (var (left, right) in tupleElements)
+        {
+            var constraint = BinaryConstraint(operation.OperatorKind, state[left], state[right], isLoopCondition, visitCount);
+            if (constraint is null)
+            {
+                allEqual = false;
+            }
+            else if (operation.OperatorKind is BinaryOperatorKind.Equals && constraint.Equals(BoolConstraint.False))
+            {
+                return BoolConstraint.False;
+            }
+            else if (operation.OperatorKind is BinaryOperatorKind.NotEquals && constraint.Equals(BoolConstraint.True))
+            {
+                return BoolConstraint.True;
+            }
+        }
+        return (allEqual, operation.OperatorKind) switch
+        {
+            (true, BinaryOperatorKind.Equals) => BoolConstraint.True,
+            (true, BinaryOperatorKind.NotEquals) => BoolConstraint.False,
+            _ => null
+        };
+    }
+
+    protected override ITupleBinaryOperationWrapper Convert(IOperation operation) =>
+        ITupleBinaryOperationWrapper.FromOperation(operation);
+
+    protected override ProgramState LearnBranchingConstraint(ProgramState state, ITupleBinaryOperationWrapper operation, bool isLoopCondition, int visitCount, bool falseBranch)
+    {
+        if ((operation.OperatorKind is BinaryOperatorKind.Equals) ^ falseBranch)
+        {
+            var tupleElements = TupleElements(operation.LeftOperand, operation.RightOperand);
+            foreach (var (left, right) in tupleElements)
+            {
+                state = LearnEqualityBranchingConstraint(state, left, right, operation.OperatorKind, isLoopCondition, visitCount, falseBranch);
+            }
+        }
+        return state;
+    }
+
+    private static IEnumerable<(IOperation Left, IOperation Right)> TupleElements(IOperation leftSide, IOperation rightSide)
+    {
+        if (leftSide.AsTuple() is { } leftTuple && rightSide.AsTuple() is { } rightTuple)
+        {
+            for (var i = 0; i < leftTuple.Elements.Length; i++)
+            {
+                foreach (var pair in TupleElements(leftTuple.Elements[i], rightTuple.Elements[i]))
+                {
+                    yield return pair;
+                }
+            }
```

Comment:
This will probably look more neat with the [Zip](https://learn.microsoft.com/en-us/dotnet/api/system.linq.enumerable.zip?view=net-8.0) method.